core/exec/src/mill/exec/GroupExecution.scala

```diff
@@ -33,6 +33,40 @@ private trait GroupExecution {
   def systemExit: Int => Nothing
   def exclusiveSystemStreams: SystemStreams
   def getEvaluator: () => EvaluatorApi
+  def headerData: String
+  lazy val parsedHeaderData: Map[String, ujson.Value] = {
+    import org.snakeyaml.engine.v2.api.{Load, LoadSettings}
+    val loaded = new Load(LoadSettings.builder().build()).loadFromString(headerData)
+    // recursively convert java data structure to ujson.Value
+    val envWithPwd = env ++ Seq(
+      "PWD" -> workspace.toString,
+      "PWD_URI" -> workspace.toNIO.toUri.toString
```

Comment:
`PWD` is a bit unfortunate, as we mean the workspace dir. Should be `WORKSPACE` instead.